api/src/main/java/com/google/sps/Application.java

```diff
@@ -10,6 +10,18 @@
 
 public class Application {
   public static void main(String[] args) {
+    try {
+      Firebase.init();
+      GmailService.init();
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
What's the point of catching an exception here just to print a stack trace? this redundantly does the same thing as just letting it propagate

Reply:
what's your suggestion? My intention is to make sure the user knows what's happening, should I just print out the exception?

Reply:
this does (negligibly) the same & is already built into java
https://github.com/googleinterns/step119-2020/pull/47#discussion_r458966849

Reply:
Considering that this is the first server being booted up, wouldn't this being the first method that's called? Wouldn't this make it so that the Exception propogated to the bottom of the call stack unchecked?

Reply:
It will propagate up, which prints out the stack trace. The same behavior as e.printStackTrace but without writing extra code. Propagating a checked exception does not make it unchecked (converting into RuntimeException does).

Reply:
I used the wrong pharsing, what I mean is when the exception is propagated upwards, it's being propagated toward somewhere that handles the exception, correct? if so, since this is the first method being called, where is the exception being handled, if not in this catch block. My current assumption is that you're suggesting leave the catch block empty so that the exception can be moved elsewhere, but I can't really make sense of why

Reply:
There is a [default exception handler](https://stackoverflow.com/questions/27664104/how-default-exception-handler-work) in the JVM. Probably if the firebase or gmail init fail, we don't want the program to keep running. So we don't want to silently print and continue.

Reply:
okay cool, I understand. going to push now